test/e2e/pods.go

```diff
@@ -0,0 +1,259 @@
+		pod.Name = pod.Name + "-" + randomSuffix()
+		pod.Labels["time"] = value
+		pod.Spec.Containers[0].Ports[0].HostPort = 0
+
+		By("submitting the pod to kubernetes")
+		_, err := podClient.Create(pod)
+		if err != nil {
+			Fail(fmt.Sprintf("Failed to create pod: %v", err))
+		}
+		defer func() {
+			// We call defer here in case there is a problem with
+			// the test so we can ensure that we clean up after
+			// ourselves
+			defer GinkgoRecover()
+			podClient.Delete(pod.Name)
+		}()
+
+		By("verifying the pod is in kubernetes")
+		pods, err := podClient.List(labels.SelectorFromSet(labels.Set(map[string]string{"time": value})))
+		if err != nil {
+			Fail(fmt.Sprintf("Failed to query for pods: %v", err))
+		}
+		Expect(len(pods.Items)).To(Equal(1))
+
+		By("deleting the pod")
+		podClient.Delete(pod.Name)
+		pods, err = podClient.List(labels.SelectorFromSet(labels.Set(map[string]string{"time": value})))
+		Expect(len(pods.Items)).To(Equal(0))
+	})
+
+	It("should be updated", func() {
+		podClient := c.Pods(api.NamespaceDefault)
+
+		By("loading the pod json")
+		pod := loadPodOrDie(assetPath("api", "examples", "pod.json"))
+		value := strconv.Itoa(time.Now().Nanosecond())
+		pod.Name = pod.Name + "-" + randomSuffix()
+		pod.Labels["time"] = value
+		pod.Spec.Containers[0].Ports[0].HostPort = 0
+
+		By("submitting the pod to kubernetes")
+		_, err := podClient.Create(pod)
+		if err != nil {
+			Fail(fmt.Sprintf("Failed to create pod: %v", err))
+		}
+		defer func() {
+			By("deleting the pod")
+			defer GinkgoRecover()
+			podClient.Delete(pod.Name)
+		}()
+
+		By("verifying the pod is in kubernetes")
+		waitForPodRunning(c, pod.Name)
+		pods, err := podClient.List(labels.SelectorFromSet(labels.Set(map[string]string{"time": value})))
+		if err != nil {
+			Fail(fmt.Sprintf("Failed to query for pods: %v", err))
+		}
+		Expect(len(pods.Items)).To(Equal(1))
+
+		By("retrieving the pod")
+		podOut, err := podClient.Get(pod.Name)
+		if err != nil {
+			Fail(fmt.Sprintf("Failed to get pod: %v", err))
+		}
+
+		By("updating the pod")
+		value = "time" + value
+		pod.Labels["time"] = value
+		pod.ResourceVersion = podOut.ResourceVersion
+		pod.UID = podOut.UID
+		pod, err = podClient.Update(pod)
+		if err != nil {
+			Fail(fmt.Sprintf("Failed to update pod: %v", err))
+		}
+		waitForPodRunning(c, pod.Name)
+
+		By("verifying the updated pod")
+		pods, err = podClient.List(labels.SelectorFromSet(labels.Set(map[string]string{"time": value})))
+		Expect(len(pods.Items)).To(Equal(1))
+	})
+
+	It("should contain environment variables for services", func() {
+		// Make a pod that will be a service.
+		// This pod serves its hostname via HTTP.
+		serverPod := parsePodOrDie(`{
+		  "kind": "Pod",
+		  "apiVersion": "v1beta1",
+		  "id": "srv",
+		  "desiredState": {
+			"manifest": {
+			  "version": "v1beta1",
+			  "id": "srv",
+			  "containers": [{
+				"name": "srv",
+				"image": "kubernetes/serve_hostname",
```

Comment:
Yes it does. I think the current one is :1.1 and I encourage its use.
Note that serve_hostname (not tag, or :latest) is the oldest image and adds a newline character to the newline.
Note that serve_hostname:1.0 (based on ubuntu so big) or server_hostname:1.1 (based on scratch, so small) **do not** add a newline character after the hostname.